src/txn/txn_ckpt.c

```diff
@@ -1697,3 +1700,32 @@ __wt_checkpoint_close(WT_SESSION_IMPL *session, bool final)
 
 	return (ret);
 }
+
+/*
+ * __checkpoint_diagnostic_latency --
+ *	Optionally add a 10 second delay to a checkpoint to simulate a long
+ *	running checkpoint for debug purposes. The reason for this option is
+ *	finding	operations that can block while waiting for a checkpoint to
+ *	complete.
+ */
+static void
+__checkpoint_diagnostic_latency(WT_SESSION_IMPL *session)
+{
+#ifdef HAVE_DIAGNOSTIC
```

Comment:
I'd avoid this ifdef here now, adding a flag check in the code path shouldn't be noticeable since the code isn't performance critical.